File: 240729/n개의 숫자 중 최소/min-of-n-num.c
#include <stdio.h>

int main() {
    int n;
    scanf("%d", &n);
    int arr[n];

    // 배열 입력
    for (int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }

    int count = 0;
    int min = arr[0]; // 첫 번째 값을 min으로 초기화

    // 최소값 찾기 및 특정 조건에 따라 카운트
    for (int i = 0; i < n; i++) {
        if (arr[i] < min) {
            min = arr[i]; // 최소값 업데이트
            count = 1; // 새로운 최소값을 찾았으므로 카운트 1로 초기화
        } else if (arr[i] == min) {
            count++; // 최소값과 같은 경우 카운트 증가
        }
    }

    printf("%d %d\n", min, count); // min과 count 출력

    return 0;
}

File: 240731/n개의 숫자 중 최소/min-of-n-num.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // 입력값 저장해둘 배열
        int size = sc.nextInt();
        int[] arr = new int[size];

        // min값 넣을 변수
        int minVal = Integer.MAX_VALUE;
        
        // 입력과 동시에 minVal 갱신
        for (int i=0; i<size; i++) {
            arr[i] = sc.nextInt();
            if (arr[i] < minVal)
                minVal = arr[i];
        }

        // min에 해당하는 원소 개수 검사
        int cnt = 0;
        for (int i=0; i<size; i++) {
            if (arr[i] == minVal)
                cnt++;
        }

        // 최종 min 출력, min에 해당하는 원소 개수 출력
        System.out.printf("%d %d", minVal, cnt);
    }
}